sdks/python/container/Dockerfile

```diff
@@ -82,6 +82,72 @@ RUN  \
     # Remove pip cache.
     rm -rf /root/.cache/pip
 
+# extract the deb packages
+# RUN cd /tmp && \
+#     # Install native bindings required for dependencies.
+#     apt-get update && \
+#     apt-get download \
+#        # Required by python-snappy
+#        libsnappy-dev \
+#        # Required by pyyaml (for c bindings)
+#        libyaml-dev \
+#        # This is used to speed up the re-installation of the sdk.
+#        ccache \
+#        # Required for using Beam Python SDK on ARM machines.
+#        libgeos-dev \
+#        && \
+#     mkdir /dpkg && \
+#     for deb in *.deb; do dpkg -i $deb; dpkg --extract $deb /dpkg || exit 10; done
+
+FROM gcr.io/distroless/cc-debian12 as beam
+LABEL Author "Apache Beam <dev@beam.apache.org>"
+ARG TARGETOS
+ARG TARGETARCH
+
+# copy some commands
+COPY --from=python-base /bin /bin
+
+# ARG  CHIPSET_ARCH=x86_64-linux-gnu
+# arm
+# ARG  CHIPSET_ARCH=aarch64-linux-gnu
+
+# copy python to distroless
+COPY --from=python-base /usr/local/lib/ /usr/local/lib/
+COPY --from=python-base /usr/local/bin/python /usr/local/bin/python
+COPY --from=python-base /usr/local/bin/pip /usr/local/bin/pip
```

Comment:
yes, I can polish this later once we agree with this direction.

Reply:
Copying installed distros seems brittle; is it possible to install from sources or a different distribution?

Reply:
This is what currently-used images do: https://github.com/docker-library/python/blob/9102bdad9e7610adfac4bc42ec4bea6064d7d4ac/3.10/bookworm/Dockerfile.

Reply:
I polished the file now. We copy our installed packages to distroless. It should be stable.

Reply:
FYI. https://docs.docker.com/develop/develop-images/instructions/#add-or-copy

If you have multiple Dockerfile steps that use different files from your context, COPY them individually, rather than all at once. If a specifically required file changes, then this ensures that only that step's build cache is invalidated, forcing only that step to be run again.